homeassistant/components/bmw_connected_drive/binary_sensor.py

```diff
@@ -189,6 +189,14 @@ class BMWBinarySensorEntityDescription(
         icon="mdi:car-electric",
         value_fn=lambda v: v.fuel_and_battery.is_charger_connected,
     ),
+    BMWBinarySensorEntityDescription(
+        key="is_pre_entry_climatization_enabled",
+        name="Pre entry climatization",
+        icon="mdi:car-seat-heater",
+        value_fn=lambda v: v.charging_profile.is_pre_entry_climatization_enabled
+        if v.charging_profile
+        else False,
```

Comment:
This is a multi-line lambda, which we should not use / avoid. Instead please extract this into a method.

Reply:
These conditional statements are only for needed for mypy compatibility.
`v.charging_profile` is `Optional` and can be `None`.

However this will not happen in HA as we're only allowing a sensor to be created if it is actually supported.

How would you like this to be handled? Additional functions, removing the optional from the library, or just adding a type ignore?